pkg/test/environment.go

```diff
@@ -768,11 +768,13 @@ func (e *Environment) StartInstrumentationServer(ctx context.Context) (int, chan
 		MaxHeaderBytes: 1 << 20,
 	}
 	done := make(chan bool)
-	waitctx.Go(e.ctx, func() {
-		err := autoInstrumentationServer.ListenAndServe()
-		if err != http.ErrServerClosed {
-			panic(err)
-		}
+	waitctx.Permissive.Go(e.ctx, func() {
+		go func() {
```

Comment:
It might be better to refactor the server's ListenAndServe to accept a context, then this section can be simplified.

Reply:
`ListenAndServe` is standard library here though, so I'm not sure how to achieve passing in a context there

Reply:
oops, didn't see that. this is fine, but I would change `chan bool` to `chan struct{}` and compare the error with `errors.Is`

Reply:
Removed that functionality rn, since i stopped using it anyways for the slo tests